scripts/generate_upstream_patch.sh

```diff
@@ -0,0 +1,138 @@
+#!/bin/bash
+
+set -eo pipefail
+
+cleanup() {
+  rm -rf "$1"
+  echo "cleaned up $1"
+}
+
+print_help() {
+  echo "Usage:"
+  echo "	$0 [-p <tag|branch|commit-hash>] [-w <tag|branch|commit-hash>]"
+  echo "	without any parameter, the script will generate upstream patch for both pallets and tee-worker"
+  echo "		-p specify the tag|branch|commit-hash for upstream pallets"
+  echo "		-w specify the tag|branch|commit-hash for upstream worker"
+}
+
+# This function generates a patch for the diffs between commit-A and commit-B
+# of the upstream repo, where
+# commit-A: the commit recorded in ./<TARGET_DIR>/upstream_commit
+# commit-B: the HEAD commit of upstream master or a given commit
+#
+# A patch will be generated under ./<TARGET_DIR>/upstream.patch
+generate_upstream_patch() {
+	local TARGET_DIR=$1
+	local UPSTREAM_NAME=$2
+	local UPSTREAM_URL=$3
```

Comment:
Actually these three args can be consolidated into one, `worker` or `pallets`, only TARGET_DIR needs to be adjusted. 

If that's the case, we don't need `getopts` handling either:
```
generate_upstream_patch.sh <pallets|worker> [upstream-hash]
```
is simple enough.

But I'm also fine if we leave as it is

Reply:
maybe `getopts` is still needed. Here it is designed to handle: `pallets only`, `worker only`, `pallets and worker`.